### PR TITLE
Fix scene navigation

### LINKED ISSUE #4554: WebotsJS: pick object does not work at long range
When the viewpoint is "too far" from an object, it is not possible to select it or rotate around it

### PR DESCRIPTION
**Pull request by BenjaminDeleze**
**Description**
fix #4554 and add a `toString` method for WbMatrix4

### CODE REVIEW

File: resources/web/wwi/MouseEvents.js [hunk 1]
```diff
@@ -384,8 +376,8 @@ export default class MouseEvents {
 
         // Updates camera's position and orientation
         let deltaRotation = yawRotation.mul(pitchRotation);
-        let currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(rotationCenter))
-          .add(rotationCenter);
+        let currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(this._rotationCenter))
```
Comment (ad-daniel) [inline resources/web/wwi/MouseEvents.js:379]:
```suggestion
        const currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(this._rotationCenter))
```

File: resources/web/wwi/MouseEvents.js [hunk 2]
```diff
@@ -141,11 +144,10 @@ export default class MouseEvents {
       let worldUpVector = WbWorld.instance.upVector;
       let yawRotation = glm.quat(Math.cos(halfYawAngle), sinusYaw * worldUpVector.x, sinusYaw * worldUpVector.y,
         sinusYaw * worldUpVector.z);
-
       // Updates camera's position and orientation
       let deltaRotation = yawRotation.mul(pitchRotation);
-      let currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(rotationCenter))
-        .add(rotationCenter);
+      let currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(this._rotationCenter))
```
Comment (ad-daniel) [inline resources/web/wwi/MouseEvents.js:149]:
```suggestion
      const currentPosition = deltaRotation.mul(glm.vec3(position.x, position.y, position.z).sub(this._rotationCenter))
```

Comment (ad-daniel) [reply resources/web/wwi/MouseEvents.js:149]: There might be more that could be const it seems

**Review by ad-daniel — DISMISSED** [review]
It works well now, only a couple variables that might be const

File: resources/web/wwi/MouseEvents.js [hunk 3]
```diff
@@ -427,6 +419,17 @@ export default class MouseEvents {
 
     if (typeof webots.currentView.ontouchstart === 'function')
       webots.currentView.ontouchstart(event);
+
+    let pos = MouseEvents.convertMouseEventPositionToRelativePosition(canvas, this._state.x, this._state.y);
+    this.picker.pick(pos.x, pos.y);
```
Comment (omichel) [inline resources/web/wwi/MouseEvents.js:424]:
```suggestion
    const position = MouseEvents.convertMouseEventPositionToRelativePosition(canvas, this._state.x, this._state.y);
    this.picker.pick(position.x, position.y);
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by omichel — APPROVED** [review]
Thank you!

**Pull request merged**